### PR TITLE
LS-8648 - Add Access Token as header to GRPC and HTTP Tracing Requests

### PR DESCRIPTION
**Pull request by cboppert**
Pull Request to add Lightstep-Access-Token to GRPC and HTTP Tracing Requests.

For the GRPC metadata it adds a client interceptor which appends the metadata before the rpc report call is made.

Finally this updates the javadoc version so tests and installs work properly without failing.

### CODE REVIEW

File: common/pom.xml [hunk 1]
```diff
@@ -40,6 +40,9 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>3.0.0-M1</version>
+                <configuration>
+                    <failOnError>false</failOnError>
+                </configuration>
```
Comment (cboppert) [inline common/pom.xml]: This fixes issues where cloning the repo and attempting to do a clean install would fail

Comment (cboppert) [reply common/pom.xml]: https://stackoverflow.com/questions/49460751/execution-attach-javadocs-of-goal-org-apache-maven-pluginsmaven-javadoc-plugin?noredirect=1

Comment (carlosalberto) [reply common/pom.xml]: Oh, I was about to release 0.17.0 - resetting the version change number. Is there anything I can help you with?

Comment (carlosalberto) [reply common/pom.xml]: (OT 0.32.0 is now being used in `master` though ;) )

Comment (cboppert) [reply common/pom.xml]: Hey @carlosalberto How are you? I actually updated to be 0.17.1 after merging your changes

Comment (cboppert) [reply common/pom.xml]: @carlosalberto If you're familiar with the tracer code, a review would be great!

**Comment by cboppert** [comment]
This PR creates a ClientChannelInterceptor to add LightStep-Access-Token Metadata to GRPC Calls as well as adds a header to HTTP requests.

File: grpc/src/main/java/com/lightstep/tracer/grpc/GrpcClientInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.lightstep.tracer.grpc;
+
+import io.grpc.*;
+import static io.grpc.Metadata.*;
+
+public class GrpcClientInterceptor implements ClientInterceptor {
+    public static final Key<String> ACCESS_TOKEN_HEADER = Key.of("LightStep-Access-Token", ASCII_STRING_MARSHALLER);
```
[the remark this answers is not in the document]
Comment (cboppert) [reply grpc/src/main/java/com/lightstep/tracer/grpc/GrpcClientInterceptor.java]: This wouldn't match up with any of the other tracers at this point (Go or JavaScript). I was aiming for consistency. Something canonicalizes it along the way, or I can go in and make PRs for Go and JavaScript which don't have it

File: grpc/src/main/java/com/lightstep/tracer/shared/GrpcCollectorClient.java [hunk 3]
```diff
@@ -1,16 +1,11 @@
 package com.lightstep.tracer.shared;
 
-import com.google.protobuf.ByteString;
-import com.lightstep.tracer.grpc.CollectorServiceGrpc;
+import com.lightstep.tracer.grpc.*;
```
[the remark this answers is not in the document]
Comment (cboppert) [reply grpc/src/main/java/com/lightstep/tracer/shared/GrpcCollectorClient.java]: I think IntelliJ did this, I can figure out what in the grpc package I'm using and import it independently though

[the remark this answers is not in the document]
Comment (frenchfrywpepper) [reply grpc/src/main/java/com/lightstep/tracer/shared/GrpcCollectorClient.java]: You can configure IntelliJ not to do this... 
Screen Shot 2019-04-19 at 10 10 01 AM

File: common/pom.xml [hunk 4]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <artifactId>java</artifactId>
         <groupId>com.lightstep.tracer</groupId>
-        <version>0.16.2</version>
+        <version>0.17.1</version>
```
[the remark this answers is not in the document]
Comment (cboppert) [reply common/pom.xml]: Got it!

File: common/pom.xml (hunk 1 above)
Comment (frenchfrywpepper) [reply common/pom.xml]: I found that upgrading the maven-javadoc-plugin from 3.0.0-M1 to 3.1.0 would fix the issue. Possibly better than ignoring failure?

Comment (cboppert) [reply common/pom.xml]: Okay I can do that, that's fine by me

Comment (cboppert) [reply common/pom.xml]: Although it's a very tertiary failure, but yes agreed

File: okhttp/pom.xml [hunk 5]
```diff
@@ -35,6 +35,9 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>3.0.0-M1</version>
```
Comment (frenchfrywpepper) [inline okhttp/pom.xml]: Replace with 3.1.0

Comment (cboppert) [reply okhttp/pom.xml]: Ah I had done that and didn't push it I guess

**Comment by cboppert** [comment]
Updated the description @frenchfrywpepper

File: grpc/src/main/java/com/lightstep/tracer/grpc/GrpcClientInterceptor.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package com.lightstep.tracer.grpc;
+
+import io.grpc.*;
+import static io.grpc.Metadata.*;
```
Comment (carlosalberto) [inline grpc/src/main/java/com/lightstep/tracer/grpc/GrpcClientInterceptor.java]: Can we expand these ones as well?

Comment (cboppert) [reply grpc/src/main/java/com/lightstep/tracer/grpc/GrpcClientInterceptor.java]: Oh yea good catch, one moment

**Pull request merged**